### ISSUE TITLE
progress

### ISSUE THREAD

**Issue by jp-praveen** [jp-praveen/Rendezvous#22]
Respected Sir,
        I have completed the things which we discussed in the previous meeting. The transfer time vs dv plots for up to 10 hours of transfer time is in a separate folder in the optimal_debris_removal branch.

The performance comparison for the different approaches for dvmax=1km/sec is as below,
sequence1: Greedy only
           Total debris removed by running from 1st to 50th debris=141
           Net debris removed (without including the start debris)=141-50=91 (avg=1.82)  
           avg.dv=27.99216087/50=0.55984

sequence2: Greedy Heuristic with heuristic_cost(i)=dvmax/depth search sequence_length
	   Total debris removed by running from 1st to 50th debris=118
           Net debris removed (without including the start debris)=118-50=68 (avg=1.36)  
           avg.dv=19.47358085/50=0.38947

sequence3: Greedy Heuristic with heuristic_cost(i)=dvmax/outdegree(i)
           Total debris removed by running from 1st to 50th debris=141
           Net debris removed (without including the start debris)=141-50=91 (avg=1.82)  
           avg.dv=27.99216087/50=0.55984

sequence1: Greedy Heuristic with heuristic_cost(i)=dvmax/depth search sequence_length with dvmax_available
           Total debris removed by running from 1st to 50th debris=111
           Net debris removed (without including the start debris)=111-50=61 (avg=1.22)  
           avg.dv=18.30689326/50=0.36614

I have also modified the existing code (runcode_multiple.m) and made it more general. It can be used for single or multiple spacecraft debris removal. 

Thank you.
Sincerely,
Praveen

**Comment by joelgeorgem** [comment]
Can we meet tomorrow at 10 AM?

**Comment by jp-praveen** [comment]
Yes sir, I will come to your office by 10 AM tomorrow.

**Comment by jp-praveen** [comment]
Respected Sir,
            I was looking at various sources to learn about the Gooding algorithm and I found a Matlab function that has executed the same. I downloaded the Matlab file and ran the code for different debris sets and found that the output "transfer time VS dv" curve looked more or less the same (the range of dv) when compared to the graphs which were obtained using our code. The main difference was that the graphs plotted using the downloaded lambert solver is continuous unlike ours.  

Debris 2 --> Debris 14 using the Gooding algorithm
![2-14_LA](https://user-images.githubusercontent.com/55296086/69749995-b2628880-1171-11ea-8af4-c95a013e0102.jpg)

Debris 2 --> Debris 14 using the Prussing algorithm-our code
![2-14_PR](https://user-images.githubusercontent.com/55296086/69750002-b8f10000-1171-11ea-8617-6b53f9a00d4c.jpg)

Debris 1 --> Debris 16 using the Gooding algorithm
![1-16_LA](https://user-images.githubusercontent.com/55296086/69750165-22710e80-1172-11ea-9df5-0b6c04684ed0.jpg)

Debris 1 --> Debris 16 using the Prussing algorithm-our code
![1-16_PR](https://user-images.githubusercontent.com/55296086/69750190-2e5cd080-1172-11ea-8851-954dc422ceab.jpg)

I wrote the Newton-Raphson method for finding the transfer orbit, but the code converges mostly to complex values. Shall I meet you to discuss these problems?

Thank you,
Sincerely,
Praveen

**Comment by joelgeorgem** [comment]
For 2 to 14, you have performed computations till t=4*10^4. Can you do it till 20*10^4 and post the results?

**Comment by jp-praveen** [comment]
Dear Sir,
        Here are the results.

Debris 2 --> Debris 14 using the Gooding algorithm
![2-14_LA](https://user-images.githubusercontent.com/55296086/69781980-c4761280-11d5-11ea-9672-86f42a4b6ef1.jpg)

Debris 2 --> Debris 14 using the Prussing algorithm
![2-14_PR](https://user-images.githubusercontent.com/55296086/69782010-d788e280-11d5-11ea-9520-4aaf58f1be6a.jpg)

**Comment by joelgeorgem** [comment]
Strange that my other comment did not appear!

I had asked, instead of the above, for 2 to 14 transfer, coast for 1.62 days and then compute &Delta;V for different transfer times.

Looking at Table 6 of the paper, I get a feeling that coast times can greatly affect &Delta;V.

**Comment by jp-praveen** [comment]
Yes Sir, This time we are getting a close value to the dv mentioned in the paper. They got 0.0598km/sec with a transfer time of 2592 seconds. We are getting the least dv as 0.085 km/sec with a transfer time of 1900 seconds.

Debris 2 --> Debris 14 using the Gooding algorithm
![2-14_LA_1 62coast](https://user-images.githubusercontent.com/55296086/69791782-3a38a900-11eb-11ea-8db7-a3429372db80.jpg)

Debris 2 --> Debris 14 using the Prussing algorithm
![2-14_PR_1 62coast](https://user-images.githubusercontent.com/55296086/69791806-47ee2e80-11eb-11ea-8647-a61c8c9d519d.jpg)

**Comment by joelgeorgem** [comment]
So,

1. Coast time matters
2. Lowest &Delta;V usually occurs within 5000 seconds

**Comment by joelgeorgem** [comment]
Can you do a similar exercise for another transfer, the data (coast time, transfer time, and &Delta;V) for which is available from Table 6?

**Comment by jp-praveen** [comment]
Yes Sir. I took the transfer from debris 6 to debris 18. The paper has a coast time of 0.46 days (39744 sec) and transfer time of 0.17 days (14688 sec) with dv=0.099km/sec.

Initially, when I ran the code I got the lowest dv to be around 0.57 km/sec. I was checking the possible error regions in the code, and then I realized that the number of revolutions was fixed to '0' in our code. I changed the number of revolutions to 1 and then looked at the outputs. We get the lowest dv as 0.20 km/sec for a transfer time of 8050 seconds. Further increase in the number of revolutions increased the least dv.

I checked the previous data for the transfer from debris 2 to debris 14 by varying the number of revolutions. In that case, the least dv was not much changed when the number of revolutions was changed.

Debris 6 --> Debris 18 using the Gooding algorithm with the number of revolutions set to 1.
![6-18_LA_coast39744_m=1](https://user-images.githubusercontent.com/55296086/69825519-a5a76880-1235-11ea-9f3c-15cbf9315519.jpg)

Debris 6 --> Debris 18 using the Prussing algorithm with the number of revolutions set to 1.
![6-18_PR_coast39744_m=1](https://user-images.githubusercontent.com/55296086/69825577-c2dc3700-1235-11ea-8270-dda42055a7e7.jpg)

**Comment by jp-praveen** [comment]
Sir,
       I checked a few other data and the downloaded lambert solver and our solver almost give the same graph except at few places. Since the Newton Raphson method was not converging, I used Halley's method and checked the output graphs. Halley's method gave similar outputs to that when fsolve or fzero is used. For some debris transfer, the minimum dv we get is the same to that in the journal, and in some cases, the journal dv and our dv varies by around 1km/sec.

Thank you
Sincerely,
Praveen

**Comment by joelgeorgem** [comment]
Let us meet sometime this week?

**Comment by jp-praveen** [comment]
Yes Sir. When can I come?

**Comment by joelgeorgem** [comment]
Tomorrow at 10 AM?

**Comment by jp-praveen** [comment]
Yes Sir. I will be there by 10 AM.

**Comment by jp-praveen** [comment]
Dear Sir,
         Actually, I met with a mild accident yesterday and sprained my leg. Shall I come on Wednesday? Initially, I thought I could make it.

**Comment by joelgeorgem** [comment]
No problem. Take rest; get well. Come only when you have completely recovered.

**Comment by jp-praveen** [comment]
I will come tomorrow (Wednesday) at 10 AM Sir.

**Comment by jp-praveen** [comment]
Sir, I figured out the problem that was giving the discontinuity in our outputs. Now I am getting a continuous graph and same dv value as that of the downloaded lambert solver. Can I meet you sometime this week and talk about this?

**Comment by joelgeorgem** [comment]
I don't see the code modified?

**Comment by jp-praveen** [comment]
Sir, I initially got an error while pushing the changes. Now I pushed the changes and the codes are modified.
